src/social_integration.cpp

```diff
@@ -25,16 +26,23 @@
  */
 class InternalSocialIntegrationPlugin {
 public:
-	InternalSocialIntegrationPlugin(const std::string &filename, const std::string &basepath) : library(filename), external(basepath)
+	InternalSocialIntegrationPlugin(const std::string &filename, const std::string &basepath) : library(nullptr), external(basepath)
 	{
 		openttd_info.openttd_version = _openttd_revision;
+
+		if (!ValidateSignatureFile(fmt::format("{}.sig", filename))) {
```

Comment:
For development builds, it should be possible to bypass this check, and just make it continue.